### PR TITLE
Howard/ping pong changes

### PR DESCRIPTION
**Pull request by cyberhoward**
### Checklist

- [x] CI is green.
- [x] Changelog updated.

### CODE REVIEW

**Comment by CyberHoward** [comment]
Merge and incorporate https://github.com/AbstractSDK/abstract/pull/378 before merging.

File: interchain/interchain-tests/src/bin/setup_funds.rs [hunk 1]
```diff
@@ -95,7 +95,9 @@ pub fn test_send_funds() -> AnyResult<()> {
         },
     )?;
 
-    interchain.check_ibc(STARGAZE, send_funds_tx)?;
+    interchain
+        .check_ibc(STARGAZE, send_funds_tx)?
+        .into_result()?;
```
Comment (Buckram123) [inline interchain/interchain-tests/src/bin/setup_funds.rs:100]: Doesn't check_ibc already check result?

File: modules/contracts/apps/ping-pong/src/error.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    DappError(#[from] AbstractAppError),
+
+    #[error("Caller module is not a ping pong: {source_module}")]
+    NotPingPong { source_module: ModuleInfo },
+
+    #[error("Can't start ping pong with zero pongs")]
+    ZeroPongs {},
+
+    #[error("Match not found for rematch")]
+    NothingToRematch {},
```
Comment (Buckram123) [inline modules/contracts/apps/ping-pong/src/error.rs:37]: Nit: this error variant never used

File: modules/contracts/apps/ping-pong/src/msg.rs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    /// If the opponent chain should lose (block height not even), it will try to play.
+    QueryAndMaybePingPong { opponent_chain: ChainName },
+}
+
+/// App query messages
+#[cosmwasm_schema::cw_serde]
+#[derive(QueryResponses, cw_orch::QueryFns)]
+pub enum AppQueryMsg {
+    #[returns(WinsResponse)]
+    Wins {},
```
Comment (Buckram123) [inline modules/contracts/apps/ping-pong/src/msg.rs:29]: It returns both wins and loses

File: modules/contracts/apps/ping-pong/tests/ping_pong_tests.rs [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+    assert_eq!(wins.wins, 1);
+    assert_eq!(wins.losses, 1);
+
+    let wins: WinsResponse = remote_app.wins()?;
+    assert_eq!(wins.losses, 1);
+    Ok(())
+}
+
+// #[test]
+// fn successful_remote_ping_pong() -> anyhow::Result<()> {
```
Comment (Buckram123) [inline modules/contracts/apps/ping-pong/tests/ping_pong_tests.rs:181]: Let's add a test where pingpong executed on remote pingpong. Would be great way to test if `into_dest_account_id` work as intended

File: modules/contracts/apps/ping-pong/tests/ping_pong_tests.rs [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+//         WinsResponse {
+//             wins: 0
+//         }
+//     );
+
+//     Ok(())
+// }
+
+// #[test]
+// fn rematch() -> anyhow::Result<()> {
```
Comment (Buckram123) [inline modules/contracts/apps/ping-pong/tests/ping_pong_tests.rs:244]: Query and maybe pingpong never used in these tests

**Review by Buckram123 — CHANGES_REQUESTED** [review]
Cool! Few small nits and missing tests

File: interchain/interchain-tests/src/bin/setup_funds.rs (hunk 1 above)
Comment (CyberHoward) [reply interchain/interchain-tests/src/bin/setup_funds.rs:100]: Yes but it returns a `must_use` type. I brought this up to Nicolas, think cw-orch has a PR with a fix.

File: modules/contracts/apps/ping-pong/src/msg.rs (hunk 3 above)
Comment (Buckram123) [reply modules/contracts/apps/ping-pong/src/msg.rs:29]: Let's rename it to something else?

File: framework/packages/abstract-client/src/interchain/remote_application.rs [hunk 6]
```diff
@@ -109,3 +109,31 @@ impl<'a, Chain: IbcQueryHandler, IBC: InterchainEnv<Chain>, M: ContractInstance<
         Ok(())
     }
 }
+
+// ## Traits to make generated queries work
```
[the remark this answers is not in the document]
Comment (Buckram123) [reply framework/packages/abstract-client/src/interchain/remote_application.rs:113]: Yeah, totally agree. Let's replace application deref as well then

File: framework/packages/abstract-std/src/native/ibc.rs [hunk 7]
```diff
@@ -126,7 +130,28 @@ impl IbcResult {
...
+        }
+    }
+
+    /// Get execute result
+    pub fn get_execute_events(&self) -> StdResult<Vec<Event>> {
+        match &self {
+            IbcResult::Execute { result, .. } => {
+                let result = result
+                    .as_ref()
+                    .map_err(|err| StdError::generic_err(err.clone()))?;
```
[the remark this answers is not in the document]
Comment (CyberHoward) [reply framework/packages/abstract-std/src/native/ibc.rs:145]: Don't think I can because result is `&Result`, so need to reference inner result first

**Comment by CyberHoward** [comment]
@Buckram123 could you check out failing test that I pushed? Think something is still off with the infrastructure registration.

**Comment by Buckram123** [comment]
> @Buckram123 could you check out failing test that I pushed? Think something is still off with the infrastructure registration.

`successful_ping_pong_to_home_chain`?

File: framework/packages/abstract-client/src/interchain/remote_application.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (Buckram123) [reply framework/packages/abstract-client/src/interchain/remote_application.rs:113]: Replaced

**Comment by CyberHoward** [comment]
Thank you @Buckram123 !!

**Pull request merged**